### ISSUE TITLE
System.AccessViolationException in DEBUG mode

### ISSUE THREAD

**Issue by ruapho** [smapiot/Piral.Blazor.Server#10]
# Bug Report

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

- [X] Can you reproduce the problem in a [MWE](https://en.wikipedia.org/wiki/Minimal_working_example)?
- [X] Are you running the latest version?
- [X] Did you perform a search in the issues?

## Environment Details and Version

Cloned the repository and added as submodule to my existing project to gain some deeper insights. Tried with latest version and 0.4.1.
Using .NET 8 on a Windows 11 Machine.

## Description

When running the code in DEBUG mode I get a `System.AccessViolationException: "Attempted to read or write protected memory. This is often an indication that other memory is corrupt."`
The exception is thrown in the ApplicationBuilderExtensions on Line 69.

## Steps to Reproduce

1. Clone this Repository.
2. Add a Piral Server as shown here https://blazor.piral.io/getting-started/server/setup but instead of referencing nuget packages reference Piral.Blazor.Orchestrator as project reference
3. Add a (empty) MfModule

## Expected behavior

The application should start

## Actual behavior

On startup the exception is thrown

## Possible Origin / Solution

From what is guess, the implementation of `Microsoft.AspNetCore.Components.ComponentFactory.GetComponentTypeInfo` should be replaced with a custom implementation. This ensures that during dependency resolving each Module has its own dependency container.
This seems to work in release build as `*tar = *inj` is not throwing an error.
Debug builds are more complex adding some kind of overhead so the pointer locations has to be calculated. Probably this calculation may be wrong, leading the pointer accessing memory it should not...

I've not found a more "elegant" way of replacing a method during runtime. I can only think of replacing the dependency injection system altogether which may be something for piral v2 :-)

But being able to debug the orchestrator should be possible and i have no idea why it is not.

**Comment by FlorianRappl** [comment]
Unfortunately this is something that Microsoft itself needs to change. The other possibility is that we just introduce our own dependencies here - i.e., having a custom build of the Microsoft.WebAs

**Comment by FlorianRappl** [comment]
(As of right now I rather see this as an enhancement than a bug. It would be a bug if it impacts the standard / designed use case, but the intention here was never to have somebody clone this repo and use a direct reference. Happy to discuss this further though.)

**Comment by ruapho** [comment]
My actual intention was to investigate another bug (i created a sample repo here: https://github.com/ruapho/Piral.Blazor.Server.Samples.Tractor and will probably open an issue here any time soon)

As this is an MIT licensed open source project I thought there might be a way to clone the repo to actually debug something that is not working (maybe because I am using it wrong, but I have no way to find out). Maybe contributing.

**Comment by FlorianRappl** [comment]
> As this is an MIT licensed open source project I thought there might be a way to clone the repo to actually debug something that is not working (maybe because I am using it wrong, but I have no way to find out). Maybe contributing.

Yeah sure - but as this is not the use case (referencing it as a project) its not intended like that (though as remarked nothing would be against having that work as it should / hence the enhancement).

What I actually have as a setup is that I have a local watch mode and just change the local NuGet cache upon change / watch trigger. This way, I can develop the framework and see it in action + I get the same behavior that any user would get. As the pdb is included I can still have breakpoints and investigate everything.

**Comment by ruapho** [comment]
Ah ok. I'll try to setup this. Thanks.
